src/auth/containers/requireStoreAccess.tsx

```diff
@@ -13,10 +13,12 @@ const withStoreAccess = (Component: React.FC) => (props: { [key: string]: any })
 
   useEffect(() => {
     const emailDomain = email?.split('@')[1];
-    if (state === UserState.PENDING || !(emailDomain === 'ucsd.edu' || emailDomain === 'acmucsd.org')) {
-      redirectHome();
-    } else {
-      setPermitted(true);
+    if (email) {
+      if (state === UserState.PENDING || !(emailDomain === 'ucsd.edu' || emailDomain === 'acmucsd.org')) {
+        redirectHome();
+      } else {
+        setPermitted(true);
+      }
```

Comment:
How come there's no email? When does that happen?

Reply:
`requireStoreAccess` is nested within `requireAccess`, which ensures that the user is signed in and their data has loaded. By checking that the email is defined, it ensures that we don't throw an invalid email error before the user profile has loaded.

Reply:
Yep, makes sense!